src/Bicep.Core/Diagnostics/IDiagnosticExtensions.cs

```diff
@@ -10,5 +11,61 @@ public static bool CanBeSuppressed(this IDiagnostic diagnostic)
         {
             return !(diagnostic.Level == DiagnosticLevel.Error && diagnostic is not AnalyzerDiagnostic);
         }
+
+        public static ErrorDiagnostic? AsErrorDiagnostic(this IDiagnostic diagnostic)
```

Comment:
TBH I think it's worth just killing the `ErrorDiagnostic` / `FixableErrorDiagnostic` class hierarchy, although that's obviously a pretty big refactor.

Reply:
I opened #13094 to track